File: src/pomodoro/RingSound.java
package pomodoro;

import java.applet.Applet;
import java.applet.AudioClip;

public class RingSound {
	private final AudioClip ringSound1 = Applet.newAudioClip(getClass().getResource("/resources/ring.wav"));
	private final AudioClip ringSound2 = Applet.newAudioClip(getClass().getResource("/resources/ring2.wav"));
	private final AudioClip ringSound3 = Applet.newAudioClip(getClass().getResource("/resources/ring3.wav"));

	public void play(int ringVolume) {
		if (ringVolume == 0) return;
		AudioClip audioClip;
		if (ringVolume == 1) audioClip = ringSound1;
		else if (ringVolume == 2) audioClip = ringSound2;
		else if (ringVolume == 3) audioClip = ringSound3;
		else throw new IllegalStateException();
		audioClip.play();
	}
}

File: src/pomodoro/RingSound.kt
package pomodoro

import java.applet.Applet

class RingSound {
    fun play(ringVolume: Int) {
        val audioClip = when (ringVolume) {
            0 -> null
            1 -> ringSound1
            2 -> ringSound2
            3 -> ringSound3
            else -> error("")
        }
        audioClip?.play()
    }

    companion object {
        private val ringSound1 = loadSound("/ring.wav")
        private val ringSound2 = loadSound("/ring2.wav")
        private val ringSound3 = loadSound("/ring3.wav")

        private fun loadSound(filePath: String) = Applet.newAudioClip(RingSound::class.java.getResource(filePath))
    }
}
